src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1349,13 +1350,21 @@ private function getServiceCall($id, Reference $reference = null)
      */
     private function camelize($id)
     {
+        if (isset($this->existingNames[$id])) {
+            return $this->existingNames[$id];
+        }
+
         $name = Container::camelize($id);
+        $uniqueName = $name = preg_replace('/[^a-zA-Z0-9_\x7f-\xff]/', '', $name);
+        $prefix = 1;
 
-        if (!preg_match('/^[a-zA-Z0-9_\x7f-\xff]+$/', $name)) {
-            throw new InvalidArgumentException(sprintf('Service id "%s" cannot be converted to a valid PHP method name.', $id));
+        while (in_array($uniqueName, $this->existingNames)) {
```

Comment:
We should introduce another array that stores the used service ids as keys to be able to perform this check in `O(1)` (`in_array()` searches the array in `O(n)`). So we could end up with something like `$serviceIdToMethodNameMap` which replaces the current `$existingNames` array and introduce a second array named something like `$usedMethodNames` (or something similar) where the unique name is the key with `true` being the assigned value.